lib/facter/gluster.rb

```diff
@@ -24,33 +33,23 @@
     end
   end
   output = Facter::Util::Resolution.exec("#{binary} peer status")
-  peer_count = $1.to_i if output =~ /^Number of Peers: (\d+)$/
-    if peer_count > 0 then
-      peer_list = output.scan(/^Hostname: (.+)$/).flatten.join(',')
-      # note the stderr redirection here
-      # `gluster volume list` spits to stderr :(
-      output = Facter::Util::Resolution.exec("#{binary} volume list 2>&1")
-      if output != "No volumes present in cluster" then
-        output.split.each do |vol|
-          info = Facter::Util::Resolution.exec("#{binary} volume info #{vol}")
-          vol_status = $1 if info =~ /^Status: (.+)$/
-            bricks = info.scan(/^Brick[^:]+: (.+)$/).flatten
-          volume_bricks[vol] = bricks
-          options = info.scan(/^(\w+\.[^:]+: .+)$/).flatten
-          if options then
-            volume_options[vol] = options
-          end
-          if vol_status == "Started" then
-            # if `gluster volume status` fails for some reason, it spits to stderr,
-            # so we suppress that.
-            status = Facter::Util::Resolution.exec("#{binary} volume status #{vol} 2>/dev/null")
-            if status =~ /^Brick/ then
-              volume_ports[vol] = status.scan(/^Brick [^\t]+\t+(\d+)/).flatten.uniq.sort
-            end
-          end
-        end
+  peer_count = Regexp.last_match(1).to_i if output =~ /^Number of Peers: (\d+)$/
+  if peer_count > 0
+    peer_list = output.scan(/^Hostname: (.+)$/).flatten.join(',')
+    volume_list.each do |vol|
+      info = Facter::Util::Resolution.exec("#{binary} volume info #{vol}")
+      vol_status = Regexp.last_match(1) if info =~ /^Status: (.+)$/
+      bricks = info.scan(/^Brick[^:]+: (.+)$/).flatten
+      volume_bricks[vol] = bricks
+      options = info.scan(/^(\w+\.[^:]+: .+)$/).flatten
+      volume_options[vol] = options if options
+      next unless vol_status == 'Started'
+      status = Facter::Util::Resolution.exec("#{binary} volume status #{vol} 2>/dev/null")
+      if status =~ /^Brick/
+        volume_ports[vol] = status.scan(/^Brick [^\t]+\t+(\d+)/).flatten.uniq.sort
       end
```

Comment:
this block looks sane, but I'm unable to verify it :(

(note to myself: setup private gluster)